Source/SwiftLintFramework/Rules/SwitchCaseAlignmentRule.swift

```diff
@@ -91,18 +37,163 @@ public struct SwitchCaseAlignmentRule: ASTRule, ConfigurationProviderRule {
 
         let caseLocations = caseStatements.compactMap { caseDict -> Location? in
             guard let byteOffset = caseDict.offset,
-                let (line, char) = contents.lineAndCharacter(forByteOffset: byteOffset) else {
-                    return nil
+                  let (line, char) = contents.lineAndCharacter(forByteOffset: byteOffset) else {
+                return nil
             }
+
             return Location(file: file.path, line: line, character: char)
         }
 
+        guard let firstCase = caseLocations.first,
+              let firstCaseCharacter = firstCase.character else {
+            return []
+        }
+
+        // If indent_cases is on, the first case should be indented from its containing switch.
+        if configuration.indentedCases, firstCaseCharacter <= switchCharacter {
+            return caseLocations.map(locationToViolation)
+        }
+
+        let indentation = configuration.indentedCases ? firstCaseCharacter - switchCharacter : 0
+
         return caseLocations
-            .filter { $0.character != switchCharacter }
-            .map {
-                StyleViolation(ruleDescription: type(of: self).description,
-                               severity: configuration.severity,
-                               location: $0)
-            }
+            .filter { $0.character != switchCharacter + indentation }
+            .map(locationToViolation)
+    }
+
+    private func locationToViolation(_ location: Location) -> StyleViolation {
+        // swiftlint:disable line_length
+        let reason = configuration.indentedCases ? "Case statements should be indented within their enclosing switch statement"
+                                                 : "Case statements should vertically align with their enclosing switch statement"
+        // swiftlint:enable line_length
+
+        return StyleViolation(ruleDescription: SwitchCaseAlignmentRule.description,
+                              severity: configuration.severityConfiguration.severity,
+                              location: location,
+                              reason: reason)
+    }
+}
+
+extension SwitchCaseAlignmentRule {
+    struct Examples {
+        private let indentedCasesOption: Bool
+        private let violationMarker = "↓"
```

Comment:
Not in this PR, but we should find a way to expose this (preferably not in the public API) for general use in this project.

Reply:
You Jean the examples themselves ? Expose in what way ?

Reply:
I mean the violation marker (`↓`). Not for the inline use when in triggering/nontriggering example markup (since those uses are designed to be quick to visually grok) but rather for times when we either parse or dynamically insert violation markers.

Reply:
Yeah I agree - it was a tad strange to copy-paste the marker everywhere but it wasn't available in any internal-scoped place in the module. We can look into it later for sure 👍